docs/guides/web-api-basics.md

```diff
@@ -0,0 +1,444 @@
+  .channel("#random")
+  .text(":wave: Hi from a bot written in Java!")
+  .build());
+```
+
+It's not over yet, the way to do the same can be much simpler if you use a function argument for initializing a request.
+
+```java
+ChatPostMessageResponse response = slack.methods(token).chatPostMessage(req -> req
+  .channel("#random")
+  .text(":wave: Hi from a bot written in Java!"));
+```
+
+I'm sure most people should agree the last one is the handiest and concise. We'll use this way for code snippets from here.
+
+```java
+import com.slack.api.Slack;
+import com.slack.api.methods.response.chat.ChatPostMessageResponse;
+
+Slack slack = Slack.getInstance();
+String token = System.getenv("SLACK_TOKEN");
+
+ChatPostMessageResponse response = slack.methods(token).chatPostMessage(req -> req
+  .channel("#random")
+  .text(":wave: Hi from a bot written in Java!"));
+```
+
+If you prefer Kotlin language, your code can be even more concise.
+
+```kotlin
+import com.slack.api.Slack
+
+val slack = Slack.getInstance()
+val token = System.getenv("SLACK_TOKEN")
+
+val response = slack.methods(token).chatPostMessage { it
+  .channel("#random")
+  .text(":wave: Hi from a bot written in Kotlin!") 
+}
+```
+
+### Handle Responses
+
+If you're not yet familiar with the Slack Web API response format, read the [Evaluating responses](https://api.slack.com/web#responses) guide to understand it. All Web API responses contain a JSON object, which always contains a top-level boolean property "ok", indicating success or failure.
+
+```json
+{
+  "ok": true,
+  "stuff": "This is good"
+}
+```
+
+```json
+{
+  "ok": false,
+  "error": "something_bad"
+}
+```
+
+So, you handle responses by evaluating the boolean value returned by `isOk()` method first, then dispatch the operations afterward.
+
+```java
+import com.slack.api.model.Message;
+
+ChatPostMessageResponse response = slack.methods(token).chatPostMessage(req -> req
+  .channel("C1234567")
+  .text("Write one, post anywhere"));
+if (response.isOk()) {
+  Message postedMessage = response.getMessage();
+} else {
+  String errorCode = response.getError(); // e.g., "invalid_auth", "channel_not_found"
+}
+```
+
+Not only the case you receive a successful response containing `"ok": false` property at the top level, errors can happen for many reasons. You need to be aware of the following three error patterns when using this SDK.
+
+1. Received a successful response but in its body, `"ok"` is **false** and an `"error"` such as `channel_not_found` exists
+2. Got a **java.io.IOException** thrown due to connectivity issues
+3. Got a **com.slack.api.methods.SlackApiException** thrown for an unsuccessful response
+
+To understand how to handle **1.** pattern, read [this API document](https://api.slack.com/web#evaluating_responses).
+
+As for **2.** & **3.** patterns, the **MethodsClient** may throw two types of exceptions. Applications using this library is responsible for catching and handling these exceptions properly.
```

Comment:
Suggestion: "Applications using this library is responsible for catching and handling these exceptions properly." -> "Applications are responsible for catching and handling both of these exceptions."